### PR TITLE
Some minor changes ... again

### PR DESCRIPTION
**Pull request by RedstoneDubstep**
For the explanations of the changes, please see the Changelog file. If you have any questions, suggestions, messages etc., please use comments or the reviewing feature. Thanks!

### CODE REVIEW

File: src/main/java/net/geforcemods/securitycraft/tileentity/TrophySystemTileEntity.java [hunk 1]
```diff
@@ -85,6 +90,9 @@ private Entity getTarget() {
 		// projectile types if we think of any
 		potentialTargets.addAll(world.getEntitiesWithinAABB(ArrowEntity.class, area));
 		potentialTargets.addAll(world.getEntitiesWithinAABB(FireballEntity.class, area));
+		potentialTargets.addAll(world.getEntitiesWithinAABB(DragonFireballEntity.class, area));
+		potentialTargets.addAll(world.getEntitiesWithinAABB(WitherSkullEntity.class, area));
+		potentialTargets.addAll(world.getEntitiesWithinAABB(ShulkerBulletEntity.class, area));
```
Comment (bl4ckscor3) [inline src/main/java/net/geforcemods/securitycraft/tileentity/TrophySystemTileEntity.java]:
```suggestion
		potentialTargets.addAll(world.getEntitiesWithinAABB(DamagingProjectileEntity.class, area));
```

File: Changelog v1.8.x.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 - New: More reinforced block recipes for parity with vanilla (Thanks Redstone_Dubstep!)
 - New: Fake Water/Lava Buckets can now be properly used in dispensers
 - New: Add all ownable blocks to DRAGON_IMMUNE and WITHER_IMMUNE block tags
+- New: Trophy system now destroys Shulker bullets, Dragon fireballs and Wither skulls
```
Comment (bl4ckscor3) [inline Changelog v1.8.x.md]: This is a change.

**Comment by bl4ckscor3** [comment]
What did you do to the textures to fix the models?

File: Changelog v1.8.x.md [hunk 3]
```diff
@@ -11,6 +12,10 @@
 - Fix: Incorrect version check with VersionChecker mod integration
 - Fix: Cage Trap owner collision is incorrect
```
Comment (bl4ckscor3) [inline Changelog v1.8.x.md]: Your cage trap changes are not reflected in the changelog, so I guess you can just remove "owner" from this line.

File: Changelog v1.8.x.md [hunk 4]
```diff
@@ -11,6 +12,10 @@
...
 - Fix: Inventory Scanners can connect to Inventory Scanners by different owners
+- Fix: Bouncing Betty doesn't play a sound when triggered and doesn't jump as high as expected
+- Fix: Block Mines let items and the owner go through
+- Fix: Trophy system destroys projectile even if it has already been removed
+- Fix: Textures of UBM and UOC are transparent at the sides
```
[the remark this answers is not in the document]
Comment (RedstoneDubstep) [reply Changelog v1.8.x.md]: > What did you do to the textures to fix the models?
I noticed (and I noticed it only on the UBM and the UOC, if you think more items have this issue, please tell me them) that you could see through the sides of those tools. This was because there were pixels that were almost invisible due to their high alpha value, but they were there and thus Minecraft rendered the sides of the tools like the most outside pixels of the tool, and as these pixes were in some cases invisible, the outline was, too. All I did was clearing out the invisible pixels from the item texture.

File: src/main/java/net/geforcemods/securitycraft/tileentity/TrophySystemTileEntity.java [hunk 5]
```diff
@@ -60,7 +65,7 @@ private void destroyTarget() {
 		entityBeingTargeted.remove();
 
 		if(!world.isRemote)
-			world.createExplosion(null, entityBeingTargeted.getPosX(), entityBeingTargeted.getPosY(), entityBeingTargeted.getPosZ(), 1.0F, Explosion.Mode.NONE);
+			world.createExplosion(null, entityBeingTargeted.getPosX(), entityBeingTargeted.getPosY(), entityBeingTargeted.getPosZ(), 0.1F, Explosion.Mode.NONE);
```
[the remark this answers is not in the document]
Comment (bl4ckscor3) [reply src/main/java/net/geforcemods/securitycraft/tileentity/TrophySystemTileEntity.java]: Nope, you can leave it (no need for a note in the changelog).

File: Changelog v1.8.x.md (hunk 2 above)
Comment (RedstoneDubstep) [reply Changelog v1.8.x.md]: >What did you do to the textures to fix the models? (your question below this request)

I noticed (and I noticed it only on the UBM and the UOC, if you think more items have this issue, please tell me them) that you could see through the sides of those tools. This was because there were pixels that were almost invisible due to their high alpha value, but they were there and thus Minecraft rendered the sides of the tools like the most outside pixels of the tool, and as these pixes were in some cases invisible, the outline was, too. All I did was clearing out the invisible pixels from the item texture.

File: Changelog v1.8.x.md [hunk 6]
```diff
@@ -3,14 +3,20 @@
 - New: More reinforced block recipes for parity with vanilla (Thanks Redstone_Dubstep!)
 - New: Fake Water/Lava Buckets can now be properly used in dispensers
 - New: Add all ownable blocks to DRAGON_IMMUNE and WITHER_IMMUNE block tags
+- New: Trophy System now destroys shulker bullets, dragon fireballs and wither skulls
+- Change: Bouncing Betty now jump as high as the player's head
```
Comment (bl4ckscor3) [inline Changelog v1.8.x.md]:
```suggestion
- Change: Bouncing Betty now jumps as high as the player's head
```

File: Changelog v1.8.x.md [hunk 7]
```diff
@@ -3,14 +3,20 @@
...
 - Fix: Incorrect laser removal in creative mode
 - Fix: Reinforced Glowstone and Reinforced Sea Lantern don't give off light (Thanks Redstone_Dubstep!)
 - Fix: German language fixes (Thanks Redstone_Dubstep!)
 - Fix: Incorrect version check with VersionChecker mod integration
-- Fix: Cage Trap owner collision is incorrect
+- Fix: Cage Trap collision is incorrect
 - Fix: Inventory Scanners can connect to Inventory Scanners by different owners
+- Fix: Bouncing Betty doesn't play a sound when triggered
+- Fix: Block Mines let items and the owner go through
+- Fix: Trophy System destroys projectile even if it has already been removed
```
Comment (bl4ckscor3) [inline Changelog v1.8.x.md]:
```suggestion
- Fix: Trophy System destroys projectiles even if they have already been removed
```

File: Changelog v1.8.x.md [hunk 8]
```diff
@@ -3,14 +3,20 @@
...
+- Fix: Textures of Universal Block Modifier, Universal Owner Changer and the Mine Remote Access tool are transparent at the sides
```
Comment (bl4ckscor3) [inline Changelog v1.8.x.md]:
```suggestion
- Fix: Textures of Universal Block Modifier, Universal Owner Changer and the Mine Remote Access Tool are transparent at the sides
```

I don't know of any other items at the moment, if you want you can go through the items of the mod, or we'll fix them as we find them.

File: src/main/java/net/geforcemods/securitycraft/blocks/CageTrapBlock.java [hunk 9]
```diff
@@ -47,12 +51,30 @@ public CageTrapBlock(Material material) {
...
 	public VoxelShape getCollisionShape(BlockState state, IBlockReader world, BlockPos pos, ISelectionContext ctx){
-		if(ctx instanceof EntitySelectionContext && ((EntitySelectionContext)ctx).getEntity() instanceof PlayerEntity)
-		{
+		if(ctx instanceof EntitySelectionContext) {
 			TileEntity te = world.getTileEntity(pos);
 
-			if(te instanceof IOwnable && ((IOwnable)te).getOwner().isOwner((PlayerEntity)((EntitySelectionContext)ctx).getEntity()))
-				return VoxelShapes.fullCube();
+
+
```
Comment (bl4ckscor3) [inline src/main/java/net/geforcemods/securitycraft/blocks/CageTrapBlock.java]: Remove these blank lines, they're completely unnecessary.

File: src/main/java/net/geforcemods/securitycraft/blocks/CageTrapBlock.java [hunk 10]
```diff
@@ -47,12 +51,30 @@ public CageTrapBlock(Material material) {
...
+			}
+
+			if (((EntitySelectionContext) ctx).getEntity() instanceof ItemEntity && te instanceof DisguisableTileEntity)
+			{
+				DisguisableTileEntity te1 = (DisguisableTileEntity) te;
+
+				if (!te1.getModule(CustomModules.DISGUISE).isEmpty() && (((ModuleItem)te1.getModule(CustomModules.DISGUISE).getItem()).getBlockAddons(te1.getModule(CustomModules.DISGUISE).getTag()).size() > 0))
+					return super.getCollisionShape(state, world, pos, ctx);
+
+
```
Comment (bl4ckscor3) [inline src/main/java/net/geforcemods/securitycraft/blocks/CageTrapBlock.java]: Remove these blank lines, they're completely unnecessary.

File: src/main/java/net/geforcemods/securitycraft/blocks/mines/BaseFullMineBlock.java [hunk 11]
```diff
@@ -44,7 +47,29 @@ public SoundType getSoundType(BlockState state)
...
 
 	@Override
-	public VoxelShape getCollisionShape(BlockState blockState, IBlockReader access, BlockPos pos, ISelectionContext ctx){
+	public VoxelShape getCollisionShape(BlockState state, IBlockReader world, BlockPos pos, ISelectionContext ctx)
+	{
+		if(ctx instanceof EntitySelectionContext)
+		{
+			Entity entity = ((EntitySelectionContext)ctx).getEntity();
+
+			if (entity instanceof  ItemEntity)
```
Comment (bl4ckscor3) [inline src/main/java/net/geforcemods/securitycraft/blocks/mines/BaseFullMineBlock.java]: Completely unnecessary space between `instanceof` and `ItemEntity`.

File: src/main/java/net/geforcemods/securitycraft/blocks/CageTrapBlock.java [hunk 12]
```diff
@@ -47,12 +51,30 @@ public CageTrapBlock(Material material) {
...
+			}
+
```
Comment (bl4ckscor3) [inline src/main/java/net/geforcemods/securitycraft/blocks/CageTrapBlock.java]: Unnecessary blank line.

File: src/main/java/net/geforcemods/securitycraft/blocks/CageTrapBlock.java [hunk 13]
```diff
@@ -47,12 +51,30 @@ public CageTrapBlock(Material material) {
...
+		if(ctx instanceof EntitySelectionContext) {
 			TileEntity te = world.getTileEntity(pos);
 
-			if(te instanceof IOwnable && ((IOwnable)te).getOwner().isOwner((PlayerEntity)((EntitySelectionContext)ctx).getEntity()))
-				return VoxelShapes.fullCube();
+
+
+			if (((EntitySelectionContext) ctx).getEntity() instanceof PlayerEntity) {
+				if (te instanceof IOwnable && ((IOwnable) te).getOwner().isOwner((PlayerEntity) ((EntitySelectionContext) ctx).getEntity()))
+					return super.getCollisionShape(state, world, pos, ctx);
```
Comment (bl4ckscor3) [inline src/main/java/net/geforcemods/securitycraft/blocks/CageTrapBlock.java]: This is incorrect. You need to return VoxelShapes.fullCube() when the Cage Trap is not disguised, otherwise return super.

File: src/main/java/net/geforcemods/securitycraft/tileentity/TrophySystemTileEntity.java [hunk 14]
```diff
@@ -83,8 +89,7 @@ private Entity getTarget() {
 
 		// Add all arrows and fireballs to the targets list. Could always add more
 		// projectile types if we think of any
-		potentialTargets.addAll(world.getEntitiesWithinAABB(ArrowEntity.class, area));
```
Comment (bl4ckscor3) [inline src/main/java/net/geforcemods/securitycraft/tileentity/TrophySystemTileEntity.java]: Why did you remove this?

File: src/main/java/net/geforcemods/securitycraft/blocks/mines/BaseFullMineBlock.java [hunk 15]
```diff
@@ -44,7 +47,29 @@ public SoundType getSoundType(BlockState state)
...
+				return VoxelShapes.fullCube();
+
```
Comment (bl4ckscor3) [inline src/main/java/net/geforcemods/securitycraft/blocks/mines/BaseFullMineBlock.java]: Unnecessary blank line. Should have been self-explanatory to remove this.

File: src/main/java/net/geforcemods/securitycraft/tileentity/TrophySystemTileEntity.java [hunk 16]
```diff
@@ -44,6 +45,11 @@ public void tick() {
 		if(entityBeingTargeted == null)
 			return;
 
+		if (!entityBeingTargeted.isAlive())
+		{
+			entityBeingTargeted = null;
+			return;
```
Comment (bl4ckscor3) [inline src/main/java/net/geforcemods/securitycraft/tileentity/TrophySystemTileEntity.java]:
```suggestion
			resetTarget();
			return;
```

File: src/main/java/net/geforcemods/securitycraft/tileentity/TrophySystemTileEntity.java (hunk 14 above)
Comment (RedstoneDubstep) [reply src/main/java/net/geforcemods/securitycraft/tileentity/TrophySystemTileEntity.java]: I thought that Arrows are also a subclass from DamagingProjectiles, my bad

**Comment by bl4ckscor3** [comment]
And exactly this is why you should start a PR from a different branch. Now because origin/1.15.x got merged into your 1.15.x, the changes from origin are reflected in this PR and it's now very cluttered and hard to follow. I will manually merge your changes locally and push them.

File: src/main/java/net/geforcemods/securitycraft/tileentity/TrophySystemTileEntity.java (hunk 14 above)
Comment (bl4ckscor3) [reply src/main/java/net/geforcemods/securitycraft/tileentity/TrophySystemTileEntity.java]: I didn't mark that line with my comment, be more careful of that, please.
